Libraries/LibGUI/GTextEditor.h

```diff
@@ -69,6 +69,8 @@ class GTextEditor
     bool has_selection() const { return m_selection.is_valid(); }
     String selected_text() const;
     void set_selection(const GTextRange&);
+    bool able_to_undo() const { return m_undo_stack_index < m_undo_stack.size() && !m_undo_stack.is_empty(); }
+    bool able_to_redo() const { return m_undo_stack_index > 0 && m_undo_stack[m_undo_stack_index - 1].m_undo_vector.size() > 0 && !m_undo_stack.is_empty(); }
```

Comment:
Let's call these `can_undo()` and `can_redo()` since that's the common idiom style throughout the codebase.